pkg/trace/stats/quantile/converter_test.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package quantile
+
+import (
+	"math"
+	"testing"
+
+	"github.com/DataDog/sketches-go/ddsketch"
+	"github.com/DataDog/sketches-go/ddsketch/mapping"
+	"github.com/DataDog/sketches-go/ddsketch/pb/sketchpb"
+	"github.com/DataDog/sketches-go/ddsketch/store"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/assert"
+)
+
+const relativeValueError = 0.01
+
+// fillBinsMap copies half of the bins to the map in order to ensure
+// that we test both the map and the array storage of bins.
+func fillBinsMap(s *sketchpb.DDSketch) {
+	s.PositiveValues.BinCounts = make(map[int32]float64)
+	n := len(s.PositiveValues.ContiguousBinCounts)
+	m := n / 2
+	for i, c := range s.PositiveValues.ContiguousBinCounts[m:] {
+		s.PositiveValues.BinCounts[int32(i+m)+s.PositiveValues.ContiguousBinIndexOffset] = c
+	}
+	s.PositiveValues.ContiguousBinCounts = s.PositiveValues.ContiguousBinCounts[:m]
+}
+
+// getConvertedSketchQuantiles follows this steps:
+// 1. Generates two DDSketches: OK and Errors:
+//    - Errors uses the generator function from 0 to n-1
+//    - OK uses the generator function from n to 2*n-1)
+// 2. Converts OK and errors DDSketches to hits and errors GK sketches (hits = ok + errors)
+//    - That way, hits is the distribution from 0 to 2*n-1
+// 3. Computes quantiles on the GK sketches hits and errors and returns them
+func getConvertedSketchQuantiles(t *testing.T, n int, gen func(i int) float64, testQuantiles []float64) (hits []float64, errors []float64) {
```

Comment:
I don't want to force your hand but normally, if possible, I'd expect a suite of tests for this function, since all tests rely on it. I haven't analysed it much so if you think it's not needed it's fine to leave it out. However, if at any time the tests break, it will be unfortunate to discover that it's because of a helper method...

Reply:
The function is a wrapper around the function that we test. So not sure what we can test that we are not already testing?
I thought about testing quantiles on the DDSketch before converting it, but it's not exported yet in the `sketches-go` library (I opened a PR to export it: https://github.com/DataDog/sketches-go/pull/22)